prow/cmd/deck/main.go

```diff
@@ -41,30 +43,56 @@ func main() {
 		logrus.WithError(err).Fatal("Error getting client.")
 	}
 
-	ja = &JobAgent{
+	ja := &JobAgent{
 		kc: kc,
 	}
 	ja.Start()
 
 	http.Handle("/", gziphandler.GzipHandler(http.FileServer(http.Dir("/static"))))
-	http.Handle("/data.js", gziphandler.GzipHandler(http.HandlerFunc(handleData)))
+	http.Handle("/data.js", gziphandler.GzipHandler(handleData(ja)))
+	http.Handle("/log", gziphandler.GzipHandler(handleLog(kc)))
 
 	logrus.WithError(http.ListenAndServe(":http", nil)).Fatal("ListenAndServe returned.")
 }
 
-func handleData(w http.ResponseWriter, r *http.Request) {
-	jobs := ja.Jobs()
-	jd, err := json.Marshal(jobs)
-	if err != nil {
-		logrus.WithError(err).Error("Error marshaling jobs.")
-		jd = []byte("[]")
+func handleData(ja *JobAgent) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		jobs := ja.Jobs()
+		jd, err := json.Marshal(jobs)
+		if err != nil {
+			logrus.WithError(err).Error("Error marshaling jobs.")
+			jd = []byte("[]")
+		}
+		// If we have a "var" query, then write out "var value = {...};".
+		// Otherwise, just write out the JSON.
+		if v := r.URL.Query().Get("var"); v != "" {
+			fmt.Fprintf(w, "var %s = %s;", v, string(jd))
+		} else {
+			fmt.Fprintf(w, string(jd))
+		}
+		w.Header().Set("Cache-Control", "no-cache")
 	}
-	// If we have a "var" query, then write out "var value = {...};".
-	// Otherwise, just write out the JSON.
-	if v := r.URL.Query().Get("var"); v != "" {
-		fmt.Fprintf(w, "var %s = %s;", v, string(jd))
-	} else {
-		fmt.Fprintf(w, string(jd))
+}
+
+type logClient interface {
+	GetLog(name string) ([]byte, error)
+}
+
+// TODO(spxtr): Cache, rate limit, and limit which pods can be logged.
```

Comment:
I'd like this to be limited to pods for actual jobs *before* we deploy it. Showing logs for other pods might inadvertently leak secrets. On the other hand, it might be infeasible to guess what any given pods name is.